servers/src/main/java/tachyon/worker/netty/DataServerHandler.java

```diff
@@ -15,42 +15,72 @@
 
 package tachyon.worker.netty;
 
-import io.netty.channel.ChannelFuture;
-import io.netty.channel.ChannelFutureListener;
-import io.netty.channel.ChannelHandler;
-import io.netty.channel.ChannelHandlerContext;
-import io.netty.channel.ChannelInboundHandlerAdapter;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import com.google.common.base.Preconditions;
 
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
+
 import tachyon.Constants;
+import tachyon.conf.TachyonConf;
 import tachyon.worker.BlockHandler;
 import tachyon.worker.BlocksLocker;
+import tachyon.worker.netty.protocol.RPCBlockRequest;
+import tachyon.worker.netty.protocol.RPCBlockResponse;
+import tachyon.worker.netty.protocol.RPCMessage;
+import tachyon.worker.netty.protocol.buffer.DataBuffer;
+import tachyon.worker.netty.protocol.buffer.DataByteBuffer;
+import tachyon.worker.netty.protocol.buffer.DataFileChannel;
 import tachyon.worker.tiered.StorageDir;
 
 /**
  * This class has the main logic of the read path to process
- * {@link tachyon.worker.netty.BlockRequest} messages and return
- * {@link tachyon.worker.netty.BlockResponse} messages.
+ * {@link tachyon.worker.netty.protocol.RPCRequest} messages and return
+ * {@link tachyon.worker.netty.protocol.RPCResponse} messages.
  */
 @ChannelHandler.Sharable
-public final class DataServerHandler extends ChannelInboundHandlerAdapter {
+public final class DataServerHandler extends SimpleChannelInboundHandler<RPCMessage> {
   private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
 
   private final BlocksLocker mLocker;
+  private final TachyonConf mTachyonConf;
+  private final FileTransferType mTransferType;
 
-  public DataServerHandler(final BlocksLocker locker) {
+  public DataServerHandler(final BlocksLocker locker, TachyonConf tachyonConf) {
     mLocker = locker;
+    mTachyonConf = tachyonConf;
+    mTransferType =
+        mTachyonConf.getEnum(Constants.WORKER_NETTY_FILE_TRANSFER_TYPE, FileTransferType.TRANSFER);
   }
 
   @Override
-  public void channelRead(final ChannelHandlerContext ctx, final Object msg) throws Exception {
-    // pipeline will make sure this is true
-    final BlockRequest req = (BlockRequest) msg;
+  public void channelRead0(final ChannelHandlerContext ctx, final RPCMessage msg) throws Exception {
```

Comment:
The parent method throws Exception, but should we make it less generic since we have an implementation?